ethsigner/src/main/java/tech/pegasys/ethsigner/requesthandler/sendtransaction/EthTransaction.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.requesthandler.sendtransaction;
+
+import tech.pegasys.ethsigner.jsonrpc.SendTransactionJsonParameters;
+
+import java.math.BigInteger;
+import java.util.List;
+
+import com.google.common.base.MoreObjects;
+import org.web3j.crypto.RawTransaction;
+import org.web3j.crypto.Sign.SignatureData;
+import org.web3j.crypto.TransactionEncoder;
+import org.web3j.rlp.RlpEncoder;
+import org.web3j.rlp.RlpList;
+import org.web3j.rlp.RlpType;
+
+public class EthTransaction implements Transaction {
+  private final SendTransactionJsonParameters sendTransactionJsonParameters;
+  private final RawTransactionBuilder rawTransactionBuilder;
+
+  public EthTransaction(final SendTransactionJsonParameters sendTransactionJsonParameters) {
+    this.sendTransactionJsonParameters = sendTransactionJsonParameters;
+    this.rawTransactionBuilder = RawTransactionBuilder.from(sendTransactionJsonParameters);
+  }
+
+  @Override
+  public void updateNonce(final BigInteger nonce) {
+    rawTransactionBuilder.updateNonce(nonce);
+  }
+
+  @Override
+  public byte[] rlpEncode(final SignatureData signatureData) {
```

Comment:
nit: YOU could break out a second rlpEncode(chainId) to align with what the TransactionEncoder does ... means the TransactionSerialiser doesn't get polluted with Web3j/Signing concepts (i.e. creating a fake signature).

Reply:
done